### PR TITLE
Replace external icon with USWDS class

### PR DESCRIPTION
**Pull request by adborden**
https://github.com/GSA/ckanext-dcat_usmetadata/issues/87 https://github.com/GSA/ckanext-dcat_usmetadata/issues/90

USWDS supports [external links](https://components.designsystem.digital.gov/components/detail/links.html) with a CSS class.

Also cleaned up some of the copy in the form description.

### CODE REVIEW

**Comment by adborden** [comment]
I ran `npx jest --updateSnapshot` to update the snapshots for the jest tests, but all 13 tests failed with the error:
```
SyntaxError: /Users/aarondborden/projects/datagov/ckanext-dcat_usmetadata/metadata-app/src/components/MetadataForm/index.test.js: Support for the experimental syntax 'jsx' isn't currently enabled
```

We should include in the README how to update these.

**Comment by kmanaseryan** [comment]
@adborden try using the following command `yarn test --updateSnapshot`. Also make sure you are running this in `./metadata-app` directory. As for adding this in the README, agree, created #90 for this.

**Comment by adborden** [comment]
Force pushed to resolve conflicts.

This is ready for review. I didn't include any build artifacts because I wasn't sure if I should simply `git add ckanext/dcat_usmetadata/public/js` or there was something else to do. I left this as a place holder in the README.

There were a few hiccups in getting the app going and fixing this issue. I opened a handful of new issues around docs and making sure we have everything in place to meet the [Definition of Done](https://github.com/GSA/datagov-ckan-multi/blob/master/CONTRIBUTING.md#exit-criteria-7).

**Comment by kmanaseryan** [comment]
I've tested this locally the icons/links looks nice. Aside from this have the following notes:

1. Nice to have better documented README, but I would create a separate PR for this just keep this more easy to review.  I've linked #90 to close once this gets merged.
2. Have you been able to commit without `--no-verify`? When I've tried to commit the resolved conflicts I got the following error:
 ```

✖ 30 problems (30 errors, 0 warnings)
  21 errors and 0 warnings potentially fixable with the `--fix` option.
```

**Review by kmanaseryan — COMMENTED** [review]
I've tested this locally the icons/links looks nice. Aside from this have the following notes:

1. Nice to have better documented README, but I would create a separate PR for this just keep this more easy to review.  I've linked #90 to close once this gets merged.
2. Have you been able to commit without `--no-verify`? When I've tried to commit the resolved conflicts I got the following error:
 ```

✖ 30 problems (30 errors, 0 warnings)
  21 errors and 0 warnings potentially fixable with the `--fix` option.
```

File: metadata-app/src/components/AdditionalMetadata/index.js [hunk 1]
```diff
@@ -32,14 +32,11 @@ const helpTexts = {
...
-      about Exhibit 53 documents at{' '}
-      <a
-        target="_blank"
-        rel="noopener noreferrer"
-        href="https://www.whitehouse.gov/wp-content/uploads/2018/06/fy-2020-it-budget-guidance.pdf"
-      >
-        this pdf <LinkIcon height="15px" />
-      </a>
+      about Exhibit 53 documents at
+      {' '}
```
Comment (kmanaseryan) [inline metadata-app/src/components/AdditionalMetadata/index.js:36]: I think this is extra

File: metadata-app/src/components/AdditionalMetadata/index.js [hunk 2]
```diff
@@ -71,14 +68,11 @@ const AdditionalMetadata = (props) => {
...
-          free to only answer what applies. For more information about the form fields, consult the{' '}
-          <a
-            target="_blank"
-            rel="noopener noreferrer"
-            href="https://resources.data.gov/schemas/dcat-us/v1.1/"
-          >
-            DCAT-US Schema <LinkIcon />
-          </a>
+          free to only answer what applies. For more information about the form fields, consult the
+          {' '}
```
Comment (kmanaseryan) [inline metadata-app/src/components/AdditionalMetadata/index.js:72]: here as well

File: metadata-app/src/components/Link/index.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      {props.children}
+    </a>
+  );
+};
+
+Link.propTypes = {
+  href: PropTypes.string,
+  target: PropTypes.string,
+  children: PropTypes.any, // eslint-disable-line
+};
```
Comment (kmanaseryan) [inline metadata-app/src/components/Link/index.js:33]: `rel` property is missing here

File: metadata-app/src/components/Link/index.fixture.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import React from 'react';
+
+export default <Link href="https://example.com">External link</Link>;
```
Comment (kmanaseryan) [inline metadata-app/src/components/Link/index.fixture.js:3]: `Link` is not defined here

**Review by kmanaseryan — CHANGES_REQUESTED** [review]
So there are a lot of ESlint and Prettier errors. @adborden let me know if you need any help with this. Basically if you have ESLint installed and configured in you IDE, without the commit you can see the errors. Alternatively when you run `yarn pre-commit` it will show where are those errors with explanation.

**Comment by kmanaseryan** [comment]
@adborden as for the [building the app](https://github.com/GSA/ckanext-dcat_usmetadata#building-the-app) you should run ` yarn build` or `npm run build` from the root directory. These will generate and update the necessary build files, after this you need to commit and push this changes.

**Comment by adborden** [comment]
@kmanaseryan I'm not using `--no-verify` when I commit. Sounds like I'm missing some git hooks? Are they documented in the README?

For linting, I'm running `make app-lint`. Is that not correct?

I'll try running `yarn pre-commit`, but again, I'm not seeing this in the README and I'm not sure why these checks aren't included in CI.

**Comment by kmanaseryan** [comment]
> @kmanaseryan I'm not using `--no-verify` when I commit. Sounds like I'm missing some git hooks? Are they documented in the README?

Totally agree, this should be explained in the README. Honestly I just knew about this when I tried to push the commit first time.  

> 
> For linting, I'm running `make app-lint`. Is that not correct?

Not sure about this. Since in the metadata app the prettier and eslint is configured, I see the errors in my IDE, also the git hook doesn't allow me to push if there is any error, unless I specify `--no-verify`. And yea this is also not documented 

> 
> I'll try running `yarn pre-commit`, but again, I'm not seeing this in the README and I'm not sure why these checks aren't included in CI.

File: metadata-app/src/components/AdditionalMetadata/index.js (hunk 1 above)
Comment (adborden) [reply metadata-app/src/components/AdditionalMetadata/index.js]: If this is removed, then there is no space between the text and the link. Personally I think this is a bug in our build configuration.
Screen Shot 2020-10-16 at 3 42 43 PM

Comment (adborden) [reply metadata-app/src/components/AdditionalMetadata/index.js]: Opened https://github.com/GSA/ckanext-dcat_usmetadata/issues/108 to discuss.

**Comment by adborden** [comment]
@kmanaseryan I fixed the `make app-lint` command to match `yarn pre-commit` and added it CI so that linting will be required before merges. I believe all your comments have been addressed.

File: README.md [hunk 5]
```diff
@@ -106,6 +106,21 @@ Run CKAN locally (`make up`) and get the Admin user's API Key. Add a test org fo
...
+
+
+### Update Jest snapshots
+
+Some tests render a fixture component with [Jest](https://jestjs.io/) and then
+match against a known good snapshot (HTML rendering) of the component. When you
+edit a component, you'll usually have to update the snapshot and inspect the
+diff to make sure all changes are as intended.
+
+    $ cd metadata-app && yarn test --updateSnapshot
```
Comment (kmanaseryan) [inline README.md:121]: Nice documentation

**Review by kmanaseryan — APPROVED** [review]
LGTM

**Pull request merged**